sqlglot/dialects/postgres.py

```diff
@@ -375,9 +389,16 @@ class Generator(generator.Generator):
         TABLESAMPLE_SIZE_IS_ROWS = False
         TABLESAMPLE_SEED_KEYWORD = "REPEATABLE"
         SUPPORTS_SELECT_INTO = True
-        # https://www.postgresql.org/docs/current/sql-createtable.html
+        JSON_TYPE_REQUIRED_FOR_EXTRACTION = True
         SUPPORTS_UNLOGGED_TABLES = True
 
+        JSON_PATH_MAPPING = {
+            exp.JSONPathChild: lambda n: n.name,
+            exp.JSONPathKey: lambda n: n.name,
+            exp.JSONPathRoot: lambda _: "",
+            exp.JSONPathSubscript: lambda n: generator.generate_json_path([n.this]),
```

Comment:
what happens when you get a foreign node? do you handle warnings?

Reply:
yep, check out this section: https://github.com/tobymao/sqlglot/pull/2883/files#diff-c779839b81bb6254e97ad8a6e17e9efbb09031d113a4942040fdd2cc3dcd9275R223, as well as https://github.com/tobymao/sqlglot/pull/2883/files#diff-63eb8dd82d3561bc414e3e13cf59516bd6584c5766a10072157a8a5aee4ad85fR71-R72

Reply:
shouldn't you call raise_error so that the error handling gets respected?

Reply:
For cases like duckdb that may be too aggressive by default, because you can have a json path but you can also have a json pointer path and so you don't really wanna hard fail there. Maybe the warning is a bit of a safer choice to start with, what do you think?

Reply:
there's nothing worse than producing the incorrect result. that's the ultimate sin in data. so if they want to ignore warnings and produce bad sql, they should opt in with errror_level ignore